### PR TITLE
Fix small UI problems with the compose reply screen

### LINKED ISSUE #7781: Fixes to UI for composing a reply
# Problem

Somehow during the latest batch of changes our reply UI got messed up a bit, this aims to fix it and improve how it renders on smaller screen devices

# Implementation

- [ ] remove the odd margin before usernames on quotes
- [ ] remove the border around quote

Screenshot 2019-03-27 at 10 25 58

- [ ] change the padding inside the quote container to 8 on all sides
- [ ] add a tiny-reply icon before the username, margin vertical is 1, margin right 2, color: grey
- [ ] change the margin between avatar and text to 8 and text to close button to 8
- [ ] position the close button to the top right of that container, change size to 28
- [ ] truncate the quote text after 3 lines with ellipsis, change font-size to 14px
- [ ] username text container is 18px high
- [ ] change the color of the border placed on top of the input toolbar and quote container to Light Grey

Screenshot 2019-03-27 at 10 52 02

here's the[ Figma link to the image above ](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=4195%3A42) and [Figma with reply designs](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1255%3A10438)  

# Acceptance Criteria

All fixes were implemented

# Notes

🙏 thanks

### PR DESCRIPTION
**Pull request by alexjg**
Fixes #7781 

### Summary
Implements the small design tweaks in the linked issue

status: ready 

I've attached a screenshot of this running on a Pixel 2. I haven't tested on an iDevice.

Screenshot (10 Sep 2019 00_24_36)

### CODE REVIEW

**Comment by errorists** [comment]
thanks @alexjg 
I'll take a more thorough look once the iOS build is ready, in the meantime judging by the attached screenshot the only issues I could find were:

• The reply icon is displayed at incorrect size (I'm guessing 20), it should be 16 by 16
• The close icon is using the incorrect asset, [this is the right one](https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=4094%3A1755). Not a huge difference, I wouldn't blame anyone for not spotting it :)
• Can we change the colour of the border at the top of the reply/chat toolbar to Light Grey, similar to how we use it elsewhere in the app?

Screenshot 2019-09-10 at 10 15 29

**Comment by alexjg** [comment]
@errorists That asset looks like it's not been added to the app (that is, I can't see a `close-circle` asset). I can't seem to go the the master component from Figma (getting a 404). Where should I get the asset and are there any conventions I should be aware of when adding it to the iOS and Android projects?

**Comment by errorists** [comment]
it's cool, I've [exported it here](https://www.dropbox.com/sh/olsrc85hnqnl4ca/AADvqI_LptPst3XlRsfQvgAEa?dl=0). As far as I can tell, we use PNGs, at 1x,2x,3x densities and tint it to get the correct colour. Alternatively, you can try and sample the close icon located under the Browser tab. It might be that it's made with a circular grey view and 'X' icon added on top, not sure, best to ask @flexsurfer 

Screenshot 2019-09-10 at 16 08 04

**Comment by flexsurfer** [comment]
yes it's just a view with the icon, you can find it here https://github.com/status-im/status-react/blob/d07c9c6613551a6b68ff1087276f08aa64f61a71/src/status_im/ui/screens/browser/open_dapp/views.cljs#L49

**Comment by alexjg** [comment]
Thanks for that, I've implemented it in a similar manner. I've also fixed the top border and the reply icon size. Let me know if anything else needs doing.

Screenshot_20190910-235754

**Comment by errorists** [comment]
looks good @alexjg ! 
@flexsurfer is there anything we can do to 'unstuck' the iOS build so it's ready to test?

**Comment by flexsurfer** [comment]
@alexjg could you please rebase onto develop and squash commits, thanks

**Comment by alexjg** [comment]
Done

**Comment by yenda** [comment]
@alexjg linter is complaining about your recent changes, you can use `lein cljfmt fix` to fix it.

File: src/status_im/ui/screens/chat/styles/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/styles/input/input.cljs:37]: does 0 padding and 0 margin change anything?

Comment (alexjg) [reply src/status_im/ui/screens/chat/styles/input/input.cljs:37]: Yep, it's overriding the values from the definition of `message-author-name`, which is

```clojure
(defn message-author-name [chosen?]
  {:font-size           (if chosen? 13 12)
   :font-weight         (if chosen? "500" "400")
   :padding-top         6
   :padding-left        12
   :padding-right       16
   :margin-right        12
   :text-align-vertical :center
   :color               colors/gray})
```

**Comment by alexjg** [comment]
> @alexjg linter is complaining about your recent changes, you can use `lein cljfmt fix` to fix it.

Done

**Review by yenda — APPROVED** [review]

**Comment by flexsurfer** [comment]
@errorists ios ready to test

**Comment by yenda** [comment]
@flexsurfer @errorists wasn't it tested already? can we merge?

**Comment by errorists** [comment]
In the testing I just did, the reply icon is not displayed at all. Other than that I couldn’t find any issues

**Comment by yenda** [comment]
@alexjg could you check the issue with the reply icons?

**Comment by alexjg** [comment]
I'll check tomorrow, @errorists was the failure on any particular platform?

**Comment by errorists** [comment]
Yes, iOS.

**Comment by alexjg** [comment]
@errorists unfortunately I can't replicate this on android and I won't have access to an idevice until the week commencing 30th September. Is there any chance it's to do with asset naming? I don't completely understand how the asset loading works but I notice that there's a `tiny-reply.imageset` in the `ios` project whereas it's `tiny_reply.png` in the android assets.

**Comment by errorists** [comment]
hey @alexjg . it's only a week, unless some savvy dev jumps here to help ( cc @yenda  ) I'd ask you to wait till then and try reproducing/fixing the image on iOS.

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/input/input.cljs]:
```suggestion
      [vector-icons/tiny-icon :tiny-icons/tiny-reply {:container-style style/reply-icon
```

File: src/status_im/ui/screens/chat/styles/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/styles/input/input.cljs:37]:
```suggestion
         ;; NOTE:  overriding the values from the definition of message-author-name
         :padding-left 0
```

**Review by yenda — DISMISSED** [review]
No need to wait a week I'm pretty sure it is because of the underscore

**Comment by yenda** [comment]
Hey @alexjg are you still planning on finishing this PR?
I resolved the conflict for you, but I can't commit my suggestion to fix the underscore issue because it is on your repo. Once you accept it please squash everything into one commit and push

**Comment by alexjg** [comment]
@yenda yep, I've been on holiday for the past few days, hence slow replies. I'll squash once I'm home tomorrow.

**Comment by alexjg** [comment]
@yenda I've squashed and pushed.

One thing I've noticed is that the recent gfycat changes mean that the address of the account you're responding to appears rather than the gfycat of it. I'm not sure if this is a bug or not and it definitely seems unrelated to the functionality implemented here but I thought I would mention it.

Everything should be hunky dory for merge now.

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/input/input.cljs]:
```suggestion
  (letsubs [{:keys [content from alias] :as message} [:chats/reply-message]]
```

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/input/input.cljs]:
```suggestion
        [reply-message from content]
```

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/input/input.cljs]: I can't make suggestions on this one since you didn't change it so here what it should look like:

```
(defview reply-message [alias {:keys [text name]]
```

**Comment by churik** [comment]
@yenda can you review it again?
I'll check briefly after and we can merge

**Comment by yenda** [comment]
@churik the code looks fine but the result isn't great because of the reply bug. I tried to fix it quickly but it is not that easy. As @alexjg mentions it comes from the username/identicons changes

**Comment by churik** [comment]
@yenda 
In this case please just ping me here when I can test it again.

**Comment by alexjg** [comment]
Is there anything I can do to move this forward? Or is it a case of waiting for a fix for the identicons issue which is being worked on elsewhere? I'm happy to have a look at the identicons issue myself if no one is doing so already.

**Comment by yenda** [comment]
@alexjg can you include this small change in your PR https://github.com/status-im/status-react/pull/9076/files ? It is not the right fix because the alias will show as a username but at least it won't show a public key instead of the name

**Comment by alexjg** [comment]
@yenda done

File: src/status_im/ui/screens/chat/utils.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/utils.cljs:15]: sorry one last thing are you sure this is actually necessary? what is this argument for? I can't find anywhere where it is true, I'm wondering if we couldn't remove the whole `chosen?` param in these styles fns

Comment (alexjg) [reply src/status_im/ui/screens/chat/utils.cljs:15]: Yeah it doesn't seem like it's used anywhere but I wasn't completely confident of that (given my inexperience with the codebase) so I made the smallest change I could. It certainly does seem like we could remove the `chosen?` parameter, want me to go ahead and do that?

**Comment by alexjg** [comment]
@yenda is there anything remaining to be done to get this merged? I'm keen to get it done as the gitcoin bounty attached to it expires in 2 days.

**Comment by alexjg** [comment]
I've rebased from `develop` and the indenticon issue looks to be solved.

**Comment by churik** [comment]
Tested on IOS 11.4.1. and Android 8 (LG V20):
- reply screen with long messages
- custom user pictures
- different types of chats

From my POV can be merged.
@flexsurfer can you review it please or we need to wait @yenda approve anyway?

File: src/status_im/ui/screens/chat/utils.cljs
Comment (yenda) [reply src/status_im/ui/screens/chat/utils.cljs:15]: @flexsurfer

Comment (yenda) [reply src/status_im/ui/screens/chat/utils.cljs:15]: I'm out this week ill let @flexsurfer decide

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**
